pages/en/zkapps/how-to-write-a-zkapp.mdx

```diff
@@ -374,28 +445,36 @@ When running this successfully, it just proves that the code was run with _some_
 and that the new `x` will be set to `hash(secret + 1)`.
 However, the secret itself remains private, because it can't be deduced from it's hash.
 
-### Managing State
+##### Initializing state
 
-##### Updating on-chain state
+We didn't cover how to initialize on-chain state yet. This can be done in the `deploy` method.
 
-As we saw before, on-chain state can be managed inside smart contract methods using `.get()` and `.set()`.
-
-<!-- To access on-chain state from outside a `SmartContract`, you can use `Mina.getAccount`:
+Like the constructor, `deploy` is predefined on the base `SmartContract` class.
+It will be called when you deploy your zkApp with the zkApp CLI.
+You can override this method to add initialization of your on-chain state:
 
 ```ts
-import { Mina } from "snarkyjs";
+let initialState = Field(10);
 
-let snappState = (await Mina.getAccount(snappAddress)).snapp.appState;
-``` -->
+class HelloWorld extends SmartContract {
+  @state(Field) x = State<Field>();
 
-##### Updating off-chain state
+  deploy(args: DeployArgs) {
+    super.deploy(args);
+    this.x.set(initialState);
+  }
+}
+```
 
-<Alert kind="info">
+Note that we have to pass the arguments on to `super.deploy`, which runs the default deploy logic. So, this line is required:
```

Comment:
Imo, this would read more clearly using `super.deploy()` when referring to methods.